commands/create/nodepool/command.go

```diff
@@ -0,0 +1,357 @@
+// Package nodepool implements the "show nodepool" command.
+package nodepool
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/gscliauth/config"
+	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/gsctl/client"
+	"github.com/giantswarm/gsctl/client/clienterror"
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/flags"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// Command is the cobra command for 'gsctl create nodepool'
+	Command = &cobra.Command{
+		Hidden:  false, // TODO: set to false to make this command visible once the release is out.
+		Use:     "nodepool <cluster-id>",
+		Aliases: []string{"np"},
+		// Args: cobra.ExactArgs(1) guarantees that cobra will fail if no positional argument is given.
+		Args:  cobra.ExactArgs(1),
+		Short: "Create a node pool",
+		Long: `Add a new node pool to a cluster.
+
+This command allows to create a new node pool within a cluster. Node pools
+are groups of wortker nodes sharing a common configuration. Create different
+node pools to serve workloads with different resource requirements, different
+availability zone spreading etc. Node pools are also scaled independently.
+
+Note that some attributes of node pools cannot be changed later. These are:
+
+- ID - will be generated on creation
+- Availability zone assignment
+- Instance type used
+
+When creating a node pool, all arguments (except for the cluster ID) are
+optional. Where an argument is not given, a default will be applied as
+follows:
+
+- Name: will be "Unnamed node pool <n>".
+- Availability zones: the node pool will use 1 zone selected randomly.
+- Instance type: the default instance type of the installation will be
+  used. Check 'gsctl info' to find out what that is.
+- Scaling settings: the minimum and maximum size will be set to 3,
+  meaning that autoscaling is disabled.
+
+Examples:
+
+  The simplest invocation creates a node pool with default attributes
+  in the cluster specified.
+
+    gsctl create nodepool f01r4
+
+  We recommend to set a descriptive name, to tell the node pool apart
+  from others.
+
+    gsctl create nodepool f01r4  --name "Batch jobs"
+
+  Assigning the node pool to availabilty zones can be done in several
+  ways. If you only want to ensure that several zones are used, specify
+  a number liker like this:
+
+    gsctl create nodepool f01r4 --num-availability-zones 2
+
+  To set one or several specific zones to use, give a list of zone names
+  or letters.
+
+    gsctl create nodepool f01r4 --availability-zones b,c,d
```

Comment:
This is very AWS specific, isn't it? How would this look like with different providers? I am not sure this zone expanding feature is a good thing.

Reply:
Node pools is AWS only at first, so is the documentation. We'll tackle the added complexity when we get there.

We simplify the zone labels in Happa, too, both in node pool creation and display. We also simplify in node pools details display in `gsctl show nodepool`. IMO it is a big benefit in readability and will prevent errors. Allowing the same simplification for node pool creation to me is just consequential.